### PR TITLE
Tag support

### PR DESCRIPTION
**Pull request by mbeltre**
- Support for tagging consumer app. eg: "prod"
- End to End tests.

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -6,7 +6,8 @@ Release History
 1.0.1 (unreleased)
 ++++++++++++++++++
 
-- Nothing changed yet.
+- Support for tagging consumer app. eg: "prod"
```
Comment (richardasaurus) [inline HISTORY.rst]: version number is wrong. please read http://semver.org/

File: pact_broker/cli.py [hunk 2]
```diff
@@ -56,6 +56,9 @@ def pull_pact(
 @click.option('--consumer_version', help='Consumer application version.')
 @click.option('--pact_dir', default='.', help='Directory to save Pacts to.')
 @click.option(
+    '--tag', default='dev', help='Consumer version tag. eg "prod".'
```
Comment (richardasaurus) [inline pact_broker/cli.py]: this defaults to "dev". will that work for everyone else in the world? (those with different pact setups etc)+

File: pact_broker/client.py [hunk 3]
```diff
@@ -80,6 +80,23 @@ def push_pact(self, *, pact_file, provider, consumer, consumer_version):
...
+            broker_url=self.broker_url,
+            consumer=consumer,
+            consumer_version=consumer_version,
+            tag=tag
+        )
+        response = requests.put(
+            request_url,
+            headers={'Content-Type': 'application/json'},
+            auth=self._auth
+        )
```
Comment (richardasaurus) [inline pact_broker/client.py]: missing `response.raise_for_status()`

File: tests/test_e2e.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+CONSUMER = 'Zoo App'
+PACT_FILE_PATH = 'tests/stubs/test_pact.json'
+TAG = 'prod'
+
+
+@pytest.fixture
+def broker_client():
+    return BrokerClient(
+        broker_url=settings.PACT_BROKER_URL
+    )
```
Comment (richardasaurus) [inline tests/test_e2e.py]: this can fit on a single line

File: tests/test_e2e.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+
+@pytest.mark.integration
+def test_upload_pact_with_tag_and_download(broker_client):
+    push_pact_response = broker_client.push_pact(
+        pact_file='tests/stubs/test_pact.json',
```
Comment (richardasaurus) [inline tests/test_e2e.py]: use constant `PACT_FILE_PATH`

File: pact_broker/cli.py (hunk 2 above)
Comment (mbeltre) [inline pact_broker/cli.py]: Making `tag` required parameter. It make sense to pact the consumer version every time upload a new pact.

File: HISTORY.rst [hunk 6]
```diff
@@ -3,10 +3,11 @@
 Release History
 ---------------
 
-1.0.1 (unreleased)
+2.0.0 (unreleased)
 ++++++++++++++++++
 
-- Nothing changed yet.
+- Support for tagging consumer app. eg: "prod"
```
Comment (richardasaurus) [inline HISTORY.rst]: version is still wrong? or is this a breaking change

**Pull request merged**